### PR TITLE
feat: add template functions for ipfs peer id

### LINKED ISSUE #329: Add ability to deterministically generate IPFS peer IDs
In order to set up a fully private IPFS network in capsule, we need to give nodes a 'bootstrap' list; and part of an entry in that list is a 'Peer ID', which is generated from a 'Peer ID Seed'

Add functions for generating a Peer ID seed given a node number, and also what the corresponding Peer ID itself is so we can point all of the datanodes at a 'golden' data node.

It would be even nicer if there were some network-specific random number in the Peer ID, but that should not be required if we stop any kind of IPFS auto discovery or connecting to the default peers on the internet.

### PR DESCRIPTION
**Pull request by pscott31**
closes #329

### CODE REVIEW

File: generator/datanode/templates.go [hunk 1]
```diff
@@ -22,6 +23,19 @@ type ConfigTemplateContext struct {
 	NodeSet     types.NodeSet
 }
 
+func (tc ConfigTemplateContext) DehistoryPeerIDSeed(nodeNumber int) string {
```
Comment (karlem) [inline generator/datanode/templates.go:26]: Can you please call it `GetDehistoryPeerIDSeed`? It's because we call all template function that.

Comment (pscott31) [reply generator/datanode/templates.go:26]: of course

File: net_confs/node_set_templates/default/data_node_full_external_postgres.tmpl [hunk 2]
```diff
@@ -43,4 +43,8 @@ GatewayEnabled = true
     Port = 30{{.NodeNumber}}5
 
 [DeHistory]
-  Enabled = true
\ No newline at end of file
+  Enabled = true
+  [DeHistory.Store]
+    IDSeed = "{{.GetDehistoryPeerIDSeed .NodeNumber}}"
+    BootstrapPeers = ["/ip4/127.0.0.1/tcp/4025/{{.GetDehistoryPeerID 2}}"]
```
Comment (daniel1302) [inline net_confs/node_set_templates/default/data_node_full_external_postgres.tmpl:49]: For now it is good, but in future we probably need function to get all data-node indexes :) to add them here like in the tendermint:

- https://github.com/vegaprotocol/vegacapsule/blob/main/generator/tendermint/templates.go#L89
- https://github.com/vegaprotocol/vegacapsule/blob/main/net_confs/node_set_templates/default/tendermint_full.tmpl#L19-L22

But i think for now it's good

**Review by daniel1302 — APPROVED** [review]

File: net_confs/node_set_templates/default/data_node_full_external_postgres.tmpl (hunk 2 above)
Comment (pscott31) [reply net_confs/node_set_templates/default/data_node_full_external_postgres.tmpl:49]: yes good idea

**Pull request merged**